services/testing-framework/claudie_test.go

```diff
@@ -32,9 +35,49 @@ const (
 	maxTimeoutSave = 60 * 12 // max allowed time for config to be found in the database
 )
 
+var (
+	//get env var from runtime directly so we do not pollute original envs package by unnecessary variables
+	cleanUpFlag = os.Getenv("AUTO_CLEAN_UP")
+	// interrupt error
+	errInterrupt = errors.New("interrupt")
+)
+
 // TestClaudie will start all the test cases specified in tests directory
 func TestClaudie(t *testing.T) {
 	utils.InitLog("testing-framework")
+	group, ctx := errgroup.WithContext(context.Background())
+
+	// start goroutine to check for SIGTERM in case pod is being deleted
+	// or interrupt in case local testing is aborted
+	group.Go(func() error {
+		ch := make(chan os.Signal, 1)
+		signal.Notify(ch, os.Interrupt, syscall.SIGTERM)
+		defer signal.Stop(ch)
```

Comment:
what happens when we stop being notified about signals sent to our program? I can see that we return an error a couple of lines below (after the "select-case"), where is it being handled afterwards, please?

Reply:
awesome :rocket:
cheers!